### PR TITLE
drivers: kscan: cleanup Kconfig

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

File: drivers/kscan/Kconfig.ht16k33 [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 config KSCAN_HT16K33
 	bool "HT16K33 keyscan driver"
-	depends on HT16K33_KEYSCAN
+	default y
+	depends on DT_HAS_HOLTEK_HT16K33_KEYSCAN_ENABLED
```
Comment (henrikbrixandersen) [inline drivers/kscan/Kconfig.ht16k33:7]: This one still depends on `CONFIG_ HT16K33_KEYSCAN` which is exposed by the HT16K33 LED driver.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: drivers/kscan/Kconfig.ht16k33 (hunk 1 above)
Comment (galak) [reply drivers/kscan/Kconfig.ht16k33:7]: Yeah, I was trying to understand that.  Wasn't sure what made the most sense here.

Comment (henrikbrixandersen) [reply drivers/kscan/Kconfig.ht16k33:7]: We could make it `select` the LED driver + key scan option similar as to how other "busses" are selected.

Comment (galak) [reply drivers/kscan/Kconfig.ht16k33:7]: @henrikbrixandersen do you mind if we merge this as is, and do a follow up PR after the LED PR (https://github.com/zephyrproject-rtos/zephyr/pull/48827) goes it for dealing with both sides at one time?

Comment (henrikbrixandersen) [reply drivers/kscan/Kconfig.ht16k33:7]: Fine by me.

**Review by henrikbrixandersen — APPROVED** [review]

**Pull request merged**